src/ripple/rpc/impl/LookupLedger.cpp

```diff
@@ -47,35 +64,58 @@ Status ledgerFromRequest (
 
     if (!hashValue.empty())
     {
+        if (! hashValue.isString ())
+            return {rpcINVALID_PARAMS, "ledgerHashNotString"};
+
         uint256 ledgerHash;
-        if (hashValue.isString() && ledgerHash.SetHex (hashValue.asString ()))
-            ledger = netOps.getLedgerByHash (ledgerHash);
-        else
+        if (! ledgerHash.SetHex (hashValue.asString ()))
             return {rpcINVALID_PARAMS, "ledgerHashMalformed"};
+           
+        if ((ledger = netOps.getLedgerByHash (ledgerHash)) == nullptr)
+        {
+            auto cl = netOps.getCurrentLedger ();
+            if (cl == nullptr || ! synced (netOps, cl))
```

Comment:
quibble - test the positive first just to avoid a `!` - or even go as far as to say:

    auto hasNetwork = cl && synced(synced (netOps, cl));
    return hasNetwork ? {rpcLGR_NOT_FOUND, "ledgerNotFound"}
        :  {rpcNO_NETWORK, "InsufficientNetworkMode"};